torch/_dynamo/symbolic_convert.py

```diff
@@ -1251,6 +1252,11 @@ def should_compile_partial_graph(self) -> bool:
     @break_graph_if_unsupported(push=0)
     def STORE_SUBSCR(self, inst):
         val, obj, key = self.popn(3)
+        if obj.source and obj.source.guard_source().is_nn_module():
+            module_source = get_nn_module_source(obj.source)
+            module = module_source.find_module(self)
+            NNModuleVariable.convert_module_to_unspecialized(self, module)
```

Comment:
This cannot possibly be complete. What about all of the other ways you can mutate a list?

Reply:
This isn't even close to complete. I am just exploring, hence the WIP. 

I am working on a refactor that identifies all "mutating" operations and routes them through a central fn where we can make decisions like this.

Reply:
Don't we already have this, isn't it side effects?